### ISSUE TITLE
Can't get ganache to start or open since following the installation guide

### ISSUE THREAD

**Issue by richwealthyandsuccessful** [trufflesuite/vscode-ext#125]
I followed the installation guide for Truffle VSCode:

https://trufflesuite.com/docs/vscode-ext/installation-guide/

But now I'm attempting the command `Truffle: Start Ganache` it's asking me to "select ganache server" which I don't know?

Alternatively, when I try the command `Truffle: Connect to Network` it lets me select a destination, then gives me a 4 digit number, but then it throws a red box up saying "No local ganache service running on port. Please start service or select another port"

Any ideas what's going on here?

**Comment by xhulz** [comment]
Hi @richwealthyandsuccessful,

Thank you for reaching out to us :)

I just created 2 screen recordings about `Truffle: Start Ganache` and `Truffle: Connect to Network`.

`Truffle: Start Ganache`: This command will only works once you have a Ganache instance created on your Network View Panel. To create a new Ganache instance, you can choose the command `Truffle: Create a new Network`

https://user-images.githubusercontent.com/32708942/177288895-4b7fa605-78ac-4ab1-b90d-570fabf26fab.mov

`Truffle: Connect to Network`: To connect to a Network, you need to have a Ganache instance already running in your machine. You can do this executing the command: `ganache --port 7545`. Once you have an instance up and running, you can choose the command `Truffle: Connect to Network` and put the port over there.

https://user-images.githubusercontent.com/32708942/177289499-8e13963d-ad16-4668-9562-f48b4f226de9.mov

Please, let me know if you have any further questions about this.

Cheers!

**Comment by richwealthyandsuccessful** [comment]
Hi xhulz,

Thank you very much for your swift reply and the video tutorials we're very helpful in understanding where I was going wrong here.

I will implement your advice now and confirm the results.

Speak soon

Marcus

**Comment by richwealthyandsuccessful** [comment]
Hi xhulz,

Good news, I was able to get ta little bit of progress, so I've created a new Ganache instance with the command `Truffle: Create a new Network` but then the following step whereby I right click "development" and choose "Start Ganache" results in a quick blip of data in the output, which then quickly disappears. Meanwhile the notification appears "Ganache server successfully started."

I then proceeded to continue to the next stage of your advice, executing the command: ganache --port 7545, but this through up "ganache command not found." 

I'll try and include a screen recording if that helps.

Thanks

Marcus

**Comment by richwealthyandsuccessful** [comment]
[Screen recording 2022-07-06 4.41.38 AM.webm](https://user-images.githubusercontent.com/107651170/177463549-90c80b5a-3264-45f0-96ef-c62e3c73a983.webm)

**Comment by xhulz** [comment]
Hey @richwealthyandsuccessful,

Thank you again :)

The most likely cause is that the global modules installation directory is not in your path, because when you type "ganache", your system doesn't recognize the command

Please have a look here:

 - https://stackoverflow.com/questions/69921649/command-not-found-ganache-cli
 - https://ethereum.stackexchange.com/questions/120292/unable-to-use-ganach-cli-in-linux

Let me know if you have more questions :)

Cheers!

**Comment by richwealthyandsuccessful** [comment]
Hi @xhulz 

Thank you for clarifying, I will definitely take a look at the links you've recommended and lets see if we can get any further along with this installation.

I'll update you with my progress. 

Kind Regards

Marcus

**Comment by richwealthyandsuccessful** [comment]
Hi Xhulz,

Thanks again, so we've made a little bit more progress from the last step, I read the Q&A from the links you provided and from what it appears I needed to use: 

`$sudo npm install -g ganache`
or
`npm install -D`

Then I was able to move further along with your steps. 

I followed the step "create a new network" (ganache instance) and it didn't glitch this time, instead it showed me the accounts, keys, wallet, gas price etc... 

From there I proceeded to the next step "connect to a network" by entering the command: 

`ganache --port 7545`

I then pressed "connect to a network" and chose the ganache service, afterwards it asked for the local port number, which I entered (7545), but it didn't recognise it, saying invalid port?

We are making steps in the right direction, but still a bit stuck here. What would you recommend here to keep moving forward?

[Screen recording 2022-07-09 5.21.18 AM.webm](https://user-images.githubusercontent.com/107651170/178091400-2ab9500b-b97b-4b91-b350-d91044812552.webm)

**Comment by xhulz** [comment]
Hey @richwealthyandsuccessful!

Glad to hear about your progress :)

Ok, these are two different options.

If you chose "Create a network" and got your Ganache instance running, you can now deploy contracts to that instance. You do not need to do anything else. Now, you are able to do some contract deployments on it.

About "Connect to a network", you will only use it when you have an external instance running on your machine, and you want to connect on it. For example: Open your bash and type `ganache --port 7545`, after that, go back to your VSCode and choose this option. You will be able to connect on that, because you have an external instance of Ganache running over the port 7545.

In other words, what you did in the first step (create a network) is enough to move forward with your deploys

Cheers!

**Comment by richwealthyandsuccessful** [comment]
Thanks for your help @xhulz I will take onboard what you've mentioned above.

I'm trying to deploy the contract I've been building based on a tutorial but it appears that when attempting to deploy my contract using `python deploy.py `or `python3 deploy.py` the following error is returning.

```
Traceback (most recent call last):
  File "/home/marcusdahibbert/demos/web3_py_simple_storage/web3_py_simple_storage/deploy.py", line 9, in <module>
    from dotenv import load_dotenv
ModuleNotFoundError: No module named 'dotenv'

```
my code so far is as follows...

```
from solcx import compile_standard, install_solc
import json
from web3 import web3

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

install_solc("0.6.0")
compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

with open("compiled_code.json", "w") as file:
    json.dump(compiled_sol, file)

# get bytecode
bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
    "bytecode"
]["object"]

# get abi
abi = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["abi"]

# for connecting to ganache
w3 = web3(web3.HTTPProvider("http://127.0.0.1:7545"))
chain_id = 1337
my_address = "0xa0F10af6e47b1ee36D25e6e6eaDa78D39F57a482"
private_key = "0xe98f12474d5d1763fbd4e70e34273813f7ddf31dcd23fad58ca15d9c2fb381d5"

# create the contract in python
SimpleStorage = w3.eth.contract(abi=abi, bytecode=bytecode)
print(SimpleStorage)
```

Abit stuck again unfortunately, would you happen to know what's going on here? I'm trying to find others with a similar problem too, just in case they found the solution.

**Comment by kevinbluer** [comment]
Hey @richwealthyandsuccessful, just to weigh in on this one...the above migration looks Python based which is a little different to the typical way in which you'd do the equivalent in a Truffle project (see this guide to [Running Migrations](https://trufflesuite.com/docs/truffle/getting-started/running-migrations/) for more detail).

That said, happy to try and help if we can. First thing to note is the endpoint you're deploying too (as per the screenshots provided) appears to point to `http://127.0.0.1:8545` (vs `http://127.0.0.1:7545`). Want to try updating that and see how it goes?

In addition, if you have a repo you can share that would be useful too.

**Comment by richwealthyandsuccessful** [comment]
Hi @kevinbluer thanks for giving your input here, I've followed your advice I believe and now the endpoint is deploying to `http://127.0.0.1:7545` I'm currently a little bit further ahead on this project, but still having trouble deploying the contract.

I'm trying to execute `python3 deploy.py` but facing the following error: 

```
Installing...
Deploying Contract!
Traceback (most recent call last):
  File "/home/marcusdahibbert/demos/web3_py_simple_storage/web3_py_simple_storage/deploy.py", line 76, in <module>
    tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)
  File "/home/marcusdahibbert/.local/lib/python3.9/site-packages/web3/eth.py", line 831, in send_raw_transaction
    return self._send_raw_transaction(transaction)
  File "/home/marcusdahibbert/.local/lib/python3.9/site-packages/web3/module.py", line 57, in caller
    result = w3.manager.request_blocking(method_str,
  File "/home/marcusdahibbert/.local/lib/python3.9/site-packages/web3/manager.py", line 198, in request_blocking
    return self.formatted_response(response,
  File "/home/marcusdahibbert/.local/lib/python3.9/site-packages/web3/manager.py", line 171, in formatted_response
    raise ValueError(response["error"])
ValueError: {'message': 'insufficient funds for gas * price + value', 'stack': 'Error: insufficient funds for gas * price + value\n    at TransactionPool.prepareTransaction (/usr/lib/node_modules/ganache/dist/node/1.js:2:133762)', 'code': -32003}

```
My code is as follows: 

```
_from solcx import compile_standard, install_solc
import json
from web3 import Web3

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

from dotenv import load_dotenv

load_dotenv()  # take environment variables from .env.

print("Installing...")
install_solc("0.6.0") 

# Solidity source code
compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {
                    "*": ["abi", "metadata", "evm.bytecode", "evm.bytecode.sourceMap"]
                }
            }
        },
    },
    solc_version="0.6.0",
)
with open("compile_code.json", "w") as file:
    json.dump(compiled_sol, file)


# get bytecode
bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
    "bytecode"
]["object"]

# get abi
abi = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["abi"]

# for conneting to ganache
w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:7545"))
chain_id = 1337
my_address = "0xa0F10af6e47b1ee36D25e6e6eaDa78D39F57a482"
private_key = "0xe98f12474d5d1763fbd4e70e34273813f7ddf31dcd23fad58ca15d9c2fb381d5"

# create the contract in python
SimpleStorage = w3.eth.contract(abi=abi, bytecode=bytecode)
# print(SimpleStorage)

# get the latest transactions
nonce = w3.eth.getTransactionCount(my_address)
print(nonce)
```
Not sure what's up here?  Here is the repo if that helps. 

```
git clone https://github.com/PatrickAlphaC/web3_py_simple_storage
cd web3_py_simple_storage
```

**Comment by richwealthyandsuccessful** [comment]
Hi @kevinbluer thanks for your feedback and support. I totally understand your point regarding this being out of your scope, I will definitely consider the option you mentioned above just in case i can't get this to work. 

Meanwhile I'm going to try to see if I can figure out how to set the gas price/ or tinker with my eth funds. However, I do seem to have sufficient funds already in the account, but I think the problem might be to do with the gas price itself. So I reckon that needs changing somehow from within the VSCode terminal, as i'm using the ganache VSCode extension.
